src/Symfony/Component/Serializer/Tests/Normalizer/ObjectNormalizerTest.php

```diff
@@ -956,19 +959,110 @@ public function testNormalizeWithMethodNamesSimilarToAccessors()
         ], $normalized);
     }
 
-    public function testNormalizeObjectWithBooleanPropertyAndIsserMethodWithSameName()
+    public function testNormalizeObjectWithPublicPropertyAccessorPrecedence()
     {
         $classMetadataFactory = new ClassMetadataFactory(new AttributeLoader());
         $normalizer = new ObjectNormalizer($classMetadataFactory);
 
-        $object = new ObjectWithBooleanPropertyAndIsserWithSameName();
+        $object = new ObjectWithPropertyAndAllAccessorMethods(
+            'foo',
+        );
         $normalized = $normalizer->normalize($object);
 
+        // The getter method should take precedence over all other accessor methods
         $this->assertSame([
             'foo' => 'foo',
-            'isFoo' => true,
         ], $normalized);
     }
+
+    public function testNormalizeObjectWithPropertyAndAccessorMethodsWithSameName()
+    {
+        $classMetadataFactory = new ClassMetadataFactory(new AttributeLoader());
+        $normalizer = new ObjectNormalizer($classMetadataFactory);
+
+        $object = new ObjectWithPropertyAndAccessorSameName(
+            'foo',
+            'getFoo',
+            'canFoo',
+            'hasFoo',
+            'isFoo'
+        );
+        $normalized = $normalizer->normalize($object);
+
+        // Previously, only the 'foo' property was extracted, all other properties starting with accessor verbs were
```

Comment:
Previously won't mean anything once merged, this should be reformulated to not refer to something that won't last